### DISCUSSION TITLE
copy contents from docs shortcut Space-k

### DISCUSSION THREAD

**Discussion by beckend** [helix-editor/helix#8420] [Ideas]
I just don't feel like manually type that thing in as a type so I can use it across the codebase.
A yank, and copy to to clipboard would be great, in for example VSCode, you can mark and copy text on popups, hints etc.
 
![2023-09-28T19:41:10,565985822-03:00](https://github.com/helix-editor/helix/assets/6056334/92787435-8bd2-4cd7-bbc0-1941db9e8350)

**Comment by the-mikedavis** [comment]
rust-analyzer has a code-action to insert explicit types: with the cursor over `client` you can use `<space>a` and use the "Insert explicit type ..." action.

**Reply by Antonio-Bennett** [reply]
That's nice but other lsp's might not have the same :( this would be very convenient. In neovim you can enter the docs. The equivalent here would be using the `leader + k` command again

**Reply by alvin883** [reply]
yes if we could enter a popup just like a buffer that would be really convenient, because sometimes I find myself wanting to copy just some portion of the type, or copy the example / certain explanation
